### PR TITLE
[tests-only][full-ci] Added API tests to check file versions of a file inside a shared space

### PR DESCRIPTION
**Pull request by ghost**
## Description

Added API tests for checking file versions of a file in a shared space (file shared with different role)

```feature
Scenario Outline: check the file versions of a file in a shared space as editor/manager
Scenario Outline: check the file versions of a file in a shared space as viewer
```

## Related Issue

- Part of : https://github.com/owncloud/ocis/issues/7824

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature [hunk 1]
```diff
@@ -30,3 +30,30 @@ Feature: checking file versions using file id
...
+      When user "Alice" gets the number of versions of file "/text.txt" using file-id path "/meta/<<FILEID>>/v"
+      Then the HTTP status code should be "207"
+      And the number of versions should be "1"
+      When user "Brian" tries to get the number of versions of file "/text.txt" using file-id path "/meta/<<FILEID>>/v"
+      Then the HTTP status code should be "<status_code>"
+      Examples:
+        | role    | status_code |
+        | editor  | 207         |
+        | viewer  | 403         |
+        | manager | 207         |
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature:59]: Please, split 207 and 403. then we can add more steps for 207 cases

**Comment by saw-jan** [comment]
Please, add some description. Mentioning the test scenarios added would be helpful

File: tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature [hunk 2]
```diff
@@ -15,7 +15,10 @@ Feature: checking file versions using file id
     And user "Alice" has uploaded a file inside space "Project1" with content "hello world version 1" to "text.txt"
     And we save it into "FILEID"
     And user "Alice" has uploaded a file inside space "Project1" with content "hello world version 1.1" to "text.txt"
-    And user "Alice" has created a share inside of space "Project1" with settings:
+    
+-
+  Scenario Outline: check the file versions of a shared file
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature:20]:
```suggestion
  Scenario Outline: check the file versions of a file shared from project space
```

File: tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature [hunk 3]
```diff
@@ -30,3 +33,34 @@ Feature: checking file versions using file id
       | editor | 
       | viewer |
       | all    |
+
+
+  Scenario Outline: check the file versions of a file in a shared space as editor and manager
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature:38]:
```suggestion
  Scenario Outline: editor/manager checks the file versions of a file inside a shared space
```

File: tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature [hunk 4]
```diff
@@ -30,3 +33,34 @@ Feature: checking file versions using file id
...
+    When user "Brian" tries to get the number of versions of file "/text.txt" using file-id path "/meta/<<FILEID>>/v"
+    Then the HTTP status code should be "207"
+    And the number of versions should be "1"
+    Examples:
+      | role    |
+      | editor  |
+      | manager |
+
+  
+  Scenario: check the file versions of a file in a shared space
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature:56]:
```suggestion
  Scenario: check the file versions of a file in a shared space as viewer
```

File: tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature [hunk 5]
```diff
@@ -30,3 +33,34 @@ Feature: checking file versions using file id
...
+    Given user "Alice" has created a share inside of space "Project1" with settings:
+      | path      | text.txt |
+      | shareWith | Brian    |
+      | role      | viewer   |
+    And using new DAV path
+    When user "Alice" gets the number of versions of file "/text.txt" using file-id path "/meta/<<FILEID>>/v"
+    Then the HTTP status code should be "207"
+    And the number of versions should be "1"
+    When user "Brian" tries to get the number of versions of file "/text.txt" using file-id path "/meta/<<FILEID>>/v"
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature:65]: Alice part is covered in above scenario
```suggestion
    When user "Brian" tries to get the number of versions of file "/text.txt" using file-id path "/meta/<<FILEID>>/v"
```

File: tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature [hunk 6]
```diff
@@ -30,3 +33,34 @@ Feature: checking file versions using file id
...
+  Scenario Outline: check the file versions of a file in a shared space as editor and manager
+    Given user "Alice" has created a share inside of space "Project1" with settings:
+      | path      | text.txt |
+      | shareWith | Brian    |
+      | role      | <role>   |
+    And using new DAV path
+    When user "Alice" gets the number of versions of file "/text.txt" using file-id path "/meta/<<FILEID>>/v"
+    Then the HTTP status code should be "207"
+    And the number of versions should be "1"
+    When user "Brian" tries to get the number of versions of file "/text.txt" using file-id path "/meta/<<FILEID>>/v"
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature:47]:
```suggestion
    When user "Brian" gets the number of versions of file "/text.txt" using file-id path "/meta/<<FILEID>>/v"
```

File: tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature [hunk 7]
```diff
@@ -26,7 +28,33 @@ Feature: checking file versions using file id
...
     Examples:
-      | role   | 
-      | editor | 
+      | role   |
+      | editor |
       | viewer |
       | all    |
+
+
+  Scenario Outline: check the file versions of a file in a shared space as editor/manager
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature:37]:
```suggestion
  Scenario Outline: check the versions of a file in a shared space as editor/manager
```

File: tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature [hunk 8]
```diff
@@ -26,7 +28,33 @@ Feature: checking file versions using file id
...
+    When user "Brian" gets the number of versions of file "/text.txt" using file-id path "/meta/<<FILEID>>/v"
+    Then the HTTP status code should be "207"
+    And the number of versions should be "1"
+    Examples:
+      | role    |
+      | editor  |
+      | manager |
+
+
+  Scenario: check the file versions of a file in a shared space as viewer
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature:54]:
```suggestion
  Scenario: check the versions of a file in a shared space as viewer
```

**Review by saw-jan — APPROVED** [review]
:+1: :sparkles:

**Review by v-scharf — APPROVED** [review]

**Pull request merged**